src/useFetch.ts

```diff
@@ -68,15 +72,41 @@ async function defaultParsing(response: Response) {
  * };
  * ```
  */
-export function useFetch<T = unknown, U = undefined>(
+type PaginatedRequestInfo = (pagination: { page: number; lastItem?: any }) => RequestInfo;
+
+export function useFetch<V = unknown, U = undefined, T = V>(
   url: RequestInfo,
-  options?: RequestInit & { parseResponse?: (response: Response) => Promise<T> } & Omit<
-      CachedPromiseOptions<(url: RequestInfo, options?: RequestInit) => Promise<T>, U>,
-      "abortable"
-    >,
+  options?: RequestInit & {
+    parseResponse?: (response: Response) => Promise<T>;
```

Comment:
how about accepting `mapResult` here as well? It's useful even without pagination